Packed resources

mymedia2 opened the pull request:
Dynamically load packed resources from separate file. This reduces amount of main memory required for linking stage.

mymedia2 commented:
Should I open MRs to submodules as well?

john-preston commented:
Yes, the submodule PRs should be merged first.

xvitaly commented on Telegram/CMakeLists.txt:
```diff
@@ -1519,7 +1524,11 @@ if (LINUX AND DESKTOP_APP_USE_PACKAGED)
+    install(TARGETS Telegram
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+        BUNDLE DESTINATION "${CMAKE_INSTALL_BINDIR}"
+        RESOURCE DESTINATION "${CMAKE_INSTALL_DATADIR}/TelegramDesktop"
```
Use lowercase directory name here: `${CMAKE_INSTALL_DATADIR}/telegram-desktop`.

mymedia2 replied:
Again, the name is not arbitrary here. It should match application name as known to Qt.

https://github.com/telegramdesktop/tdesktop/blob/v3.3.0/Telegram/SourceFiles/core/launcher.cpp#L288

mymedia2 committed:
Packed resources

Dynamically load packed resources from a separate file. This
reduces amount of main memory required for linking stage.
